eth/main.cpp

```diff
@@ -586,11 +585,28 @@ int main(int argc, char** argv)
 		}
 		else if ((arg == "-d" || arg == "--path" || arg == "--db-path") && i + 1 < argc)
 			dbPath = argv[++i];
-		else if ((arg == "--genesis-json" || arg == "--genesis" || arg == "--config") && i + 1 < argc)
+		else if ((arg == "--genesis-json" || arg == "--genesis") && i + 1 < argc)
 		{
 			try
 			{
-				paramsJSON = contentsString(argv[++i]);
+				string paramsJSON = contentsString(argv[++i]);
+				if (!paramsJSON.empty())
+					chainParams.loadGenesis(paramsJSON);
```

Comment:
If both genesis and config are given (in whatever order), I think it would make sense that the genesis overwrites whatever is given in the config file.